src/content/app/new-species-selector/components/selectable-genomes-table/useOrderedGenomes.ts

```diff
@@ -0,0 +1,198 @@
+    if (sortRule.sortOrder === 'asc') {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringAsc(a.scientific_name, b.scientific_name);
+      });
+    } else {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringDesc(a.scientific_name, b.scientific_name);
+      });
+    }
+  } else if (sortRule?.column === 'type') {
+    if (sortRule.sortOrder === 'asc') {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringAsc(
+          createGenomeTypeString(a),
+          createGenomeTypeString(b)
+        );
+      });
+    } else {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringDesc(
+          createGenomeTypeString(a),
+          createGenomeTypeString(b)
+        );
+      });
+    }
+  } else if (sortRule?.column === 'assembly_name') {
+    if (sortRule.sortOrder === 'asc') {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringAsc(a.assembly.name, b.assembly.name);
+      });
+    } else {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringDesc(a.assembly.name, b.assembly.name);
+      });
+    }
+  } else if (sortRule?.column === 'assembly_accession_id') {
+    if (sortRule.sortOrder === 'asc') {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringAsc(a.assembly.accession_id, b.assembly.accession_id);
+      });
+    } else {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortStringDesc(a.assembly.accession_id, b.assembly.accession_id);
+      });
+    }
+  } else if (sortRule?.column === 'coding_genes_count') {
+    if (sortRule.sortOrder === 'asc') {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortNumberAsc(a.coding_genes_count, b.coding_genes_count);
+      });
+    } else {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortNumberDesc(a.coding_genes_count, b.coding_genes_count);
+      });
+    }
+  } else if (sortRule?.column === 'contig_n50') {
+    if (sortRule.sortOrder === 'asc') {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortNumberAsc(a.contig_n50, b.contig_n50);
+      });
+    } else {
+      orderedGenomes = [...genomes].sort((a, b) => {
+        return sortNumberDesc(a.contig_n50, b.contig_n50);
+      });
+    }
+  }
+
+  return {
+    orderedGenomes,
+    sortRule,
+    changeSortRule
+  };
+};
+
+const createGenomeTypeString = ({ type, is_reference }: SelectableGenome) => {
+  const { kind = '', value = '' } = type || {};
+  const referenceString = is_reference ? 'reference' : '';
+  const combinedString = `${referenceString} ${kind} ${value}`;
+  return combinedString;
+};
+
+export const getSortOrderForColumn = (
+  column: SortableColumn,
+  sortRule: SortRule | null
+) => {
+  if (sortRule?.column === column) {
+    return sortRule.sortOrder;
+  } else {
+    return 'none';
+  }
+};
+
```

Comment:
The four functions below look generic enough to be moved to the helpers directory.

Reply:
✅